spec/controllers/communicarts_controller_spec.rb

```diff
@@ -157,6 +157,30 @@
         expect(token.used?).to eq(true)
       end
     end
-  end
 
+    context "version number" do
+      it "doesn't need to be present" do
+        put 'approval_response', approval_params_with_token
+        approval.reload
+        expect(approval).to be_approved
+        expect(flash[:error]).not_to be_present
+      end
+      it "works if version matches" do
+        params = approval_params_with_token
+        params[:version] = cart.proposal.version.to_s
+        put 'approval_response', params
+        approval.reload
+        expect(approval).to be_approved
+        expect(flash[:error]).not_to be_present
+      end
+      it "doesn't work if versions don't match" do
+        params = approval_params_with_token
+        params[:version] = (cart.proposal.version + 1).to_s
```

Comment:
Technically, wouldn't the param value only ever be less than that of the Proposal? Could see myself being confused by this later...mind changing to `cart.proposal.version - 1`?